Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -1803,8 +1823,46 @@ int CLuaPedDefs::RemovePedClothes(lua_State* luaVM)
     return 1;
 }
 
-bool CLuaPedDefs::SetPedControlState(CClientPed* const ped, const std::string control, const bool state) noexcept
+bool CLuaPedDefs::SetPedControlState(const std::variant<CClientPed*, std::string_view> first, const std::variant<std::string_view, bool> second, const std::optional<bool> third) noexcept
 {
+    CClientPed*      ped;
+    std::string_view control;
+    bool             state;
+
+    if (std::holds_alternative<CClientPed*>(first))
+    {
+        ped = std::get<CClientPed*>(first);
+
+        if (std::holds_alternative<std::string_view>(second))
+        {
+            control = std::get<std::string_view>(second);
+
+            if (third)
+                state = *third;
+            else
+                return false;
+        }
+        else if (std::holds_alternative<bool>(second))
+            state = std::get<bool>(second);
+        else
+            return false;
+    }
+    else if (std::holds_alternative<std::string_view>(first))
+    {
+        ped = CStaticFunctionDefinitions::GetLocalPlayer();
+        control = std::get<std::string_view>(first);
+
+        if (std::holds_alternative<bool>(second))
+            state = std::get<bool>(second);
+        else if (third)
+            state = *third;
+        else
+            return false;
+    }
+
+    if (!ped)
+        return false;
+
```

Comment:
It's a mess. And has no error messages for bad usage. I can write `setPedControlState("string1", "fire", false)` and get silent false. Maybe it's better to keep the old parser here. What do you think?